src/main/java/dk/kb/storage/storage/DsStorage.java

```diff
@@ -697,6 +718,65 @@ public int updateMTimeForRecord(String recordId) throws Exception {
     }
     
     
+    
+      /**
+       * Create a new entry in the mapping table. The kalturaid can be null and will be updated by a job later.
+       * 
+       * @param mapping The mapping.id must not be null. The kalturaid can be null-
+       * @throws Exception If mapping.id already exists.
+       */
+     public void createNewMapping(MappingDto mappingDto) throws Exception {
+
+        if (mappingDto.getReferenceId() == null) {
+            throw new InvalidArgumentServiceException("ReferenceId must not be null"); // TODO exception enum types, messages?
+        }
+
+        try (PreparedStatement stmt = connection.prepareStatement(createMapping);) {
+            stmt.setString(1, mappingDto.getReferenceId());
+            stmt.setString(2, mappingDto.getKalturaId()); 
+            stmt.executeUpdate();
+
+        } catch (SQLException e) {
+            String message = "SQL Exception in createNewMappingwith referenceId:" + mappingDto.getReferenceId() + " error:" + e.getMessage();
+            log.error(message);
+            throw new SQLException(message, e);
+        }
+    }
+    
+     
+     /**
+      * Create a new entry in the mapping table. The kalturaid can be null and will be updated by a job later.
+      * 
+      * @param mappingId The id key for the mapping
+      * @return The mappingDto or null if the id is mappingId is not found
+      * 
+      * @throws Exception
+      */
+    public MappingDto getMappingById(String mappingId) throws Exception {
+
+       if (mappingId == null) {
+           throw new InvalidArgumentServiceException("Id must not be null"); // TODO exception enum types, messages?
```

Comment:
As this mapping is done between ID's I think it is important to directly state which ID musn't be null

Reply:
fixed again here